src/reanimated2/Hooks.js

```diff
@@ -311,6 +310,12 @@ export function useAnimatedStyle(updater, dependencies) {
     };
   }, dependencies);
 
+  useEffect(() => {
+    return () => {
+      initRef.current = undefined;
```

Comment:
In `useAnimationStyle()` is used `maybeViewRef`. Although it is used only for web but maybe we should also clear this `ref` in `useEffect()`

Reply:
fixed 635aaec